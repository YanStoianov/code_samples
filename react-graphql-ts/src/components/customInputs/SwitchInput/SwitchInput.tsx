import {Switch} from "antd";
import SwitchInputStyles from "./SwitchInputStyles";
import {SwitchPropsInterface} from "../../../interfaces/inputInterfaces";

const SwitchInput = (props: SwitchPropsInterface) => {
    const {
        setValue,
        title,
        value
    } = props;

    return (
        <SwitchInputStyles>
            <Switch checked={value} onChange={setValue}/>
            <div>{title}</div>
        </SwitchInputStyles>
    )
}

export default SwitchInput