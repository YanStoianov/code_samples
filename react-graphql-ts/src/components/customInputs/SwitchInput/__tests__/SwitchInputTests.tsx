import React from 'react'
import {render} from '@testing-library/react';
import 'jest-styled-components'
import SwitchInput from "../SwitchInput";


describe("SwitchInputTest", () => {
    beforeAll(() => {
        Object.defineProperty(window, "matchMedia", {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
                matches: false,
                media: query,
                onchange: null,
                addListener: jest.fn(),
                removeListener: jest.fn(),
                addEventListener: jest.fn(),
                removeEventListener: jest.fn(),
                dispatchEvent: jest.fn(),
            }))
        });
    });

    test('snapshot', () => {
        const {asFragment} = render(
            <SwitchInput title={'test'} value={true} setValue={() => {
            }}/>
        );
        expect(asFragment()).toMatchSnapshot();
    })
});