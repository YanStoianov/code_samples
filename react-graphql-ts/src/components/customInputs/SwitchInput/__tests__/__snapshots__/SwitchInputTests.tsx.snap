// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchInputTest snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe ejzTqd"
  >
    <button
      aria-checked="true"
      class="ant-switch ant-switch-checked"
      role="switch"
      type="button"
    >
      <div
        class="ant-switch-handle"
      />
      <span
        class="ant-switch-inner"
      />
    </button>
    <div>
      test
    </div>
  </div>
</DocumentFragment>
`;
